app/models/pipeline_run_stage.rb

```diff
@@ -53,8 +59,26 @@ def failed?
     job_status == STATUS_FAILED
   end
 
-  def succeeded? # The whole thing completed successfully
-    db_load_status == 1
+  def last_output_in_stage
+    basename = "#{job_id}.done"
+    s3_folder = case name
+                when HOST_FILTERING_STAGE_NAME
+                  pipeline_run.host_filter_output_s3_path
+                when ALIGNMENT_STAGE_NAME
+                  pipeline_run.alignment_output_s3_path
+                when POSTPROCESS_STAGE_NAME
+                  pipeline_run.postprocess_output_s3_path
+                end
```

Comment:
This is nice and readable, but I am wondering is it more or less conventional than defining a hash?

Reply:
Not sure. Hash could be nice. I think I'm going to leave it like this for now, for the simple reason that it allows us to have everything in one place